lib/search/pagination.rb

```diff
@@ -54,7 +54,7 @@ def current_page
 
     def pagination_object
       {
-        'current_page' => [current_page, total_pages].min,
+        'current_page' => [current_page, total_viewable_pages].min,
```

Comment:
Below, on Line 77, there is this -

```
def maximum_viewable_entries
      (ENTRIES_PER_PAGE * total_viewable_pages) + (ENTRIES_PER_PAGE - 1)
    end
```

I cannot figure out why that `+ (ENTRIES_PER_PAGE - 1)` is there but it seems to be the root of an issue where page 100 which should contain results 991-1000 but instead it isn't accessible.

Reply:
@kfrz any insight on this?

Reply:
@ncksllvn any reason not to just limit the *input* to 99? Avoids taking a misunderstood calculation and changing it to another misunderstood one :)

Reply:
The logic was written so that the page sizes are flexible and can be changed, so many values are derived. I had some trouble understanding it, and tracing it to how the front-end responds. I think it could all be simpler though.

Reply:
@ncksllvn you're right - that's the reason we added the weird `-1` logic, the last page wouldn't be accessible otherwise. Albeit an edge case but that was the solution we shipped....

I'm in agreement with you folks, about simplifying/rewriting.

@johnpaulashenfelter I like the idea of limiting the input. Do we really need to have a flexible page size? Why not preload 99 results and handle any display pagination on the f/e (show xx results...) 

Though this change should be a good stop-gap but we *really* need to refactor this service.